### PR TITLE
fix(sdk): show loader right after visualizing in notebook editor for the first time

### LINKED ISSUE #47564: No loading state after clicking visualize when creating question in the sdk
When clicking on the "Visualize" button in the Notebook Editor in the InteractiveQuestion component, we don't get any feedback until the question has been visualized. We should either flip to the "Visualization" tab immediately to use its loading state, or have some loading state in the Visualize button itself.

<div>
    <a href="https://www.loom.com/share/a3b6fdeec06d49448e2a5654efee5420">
      <p>Clicking on Visualize while in the Notebook Editor takes time before the viz is shown - Watch Video</p>
    </a>
    <a href="https://www.loom.com/share/a3b6fdeec06d49448e2a5654efee5420">
      
    </a>
  </div>

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/47564

### Description

There is no feedback after clicking on "Visualize" for the first time when creating new questions. You can notice this in [Shoppy when creating questions](https://embedded-analytics-sdk-demo.metabase.com/admin/analytics/new/from-scratch) as the viz only shows up after the question finishes loading, which takes a long time on Shoppy.

This PR triggers `onApply` immediately when "Visualize" is clicked. It also fixes an issue where we show "Question not found" instead of a loading spinner in the QuestionVisualization view for the first time.

### How to verify

- Use either the `CreateQuestion` component in the Storybook story, or use the question creator in Shoppy with this PR.
- Click on Visualize
- It should switch to the visualize view right away. It will show a loading indicator.

### Demo

After clicking on Visualize when creating questions in Shoppy, it shows a loading indicator.

CleanShot 2567-11-22 at 23 53 32@2x

### Checklist

- [x] Tests have been added/updated to cover changes in this PR - E2E

### CODE REVIEW

File: e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts [hunk 1]
```diff
@@ -20,9 +20,17 @@ describeSDK("scenarios > embedding-sdk > create-question", () => {
...
-  it("can create questions via the CreateQuestion component", () => {
+  it("can create a question via the CreateQuestion component", () => {
     cy.intercept("POST", "/api/card").as("createCard");
 
+    cy.intercept("POST", "/api/dataset", req => {
+      // throttling makes the loading indicator shows up
+      // reliably when we click on visualize.
+      req.on("response", res => {
+        res.setThrottle(1000);
+      });
```
Comment (heypoom) [inline e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts:31]: I couldn't think of another way to reduce test flakiness when checking for loading indicators, let me know if you have a better idea 🙏🏻

File: e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts [hunk 2]
```diff
@@ -20,9 +20,17 @@ describeSDK("scenarios > embedding-sdk > create-question", () => {
...
-  it("can create questions via the CreateQuestion component", () => {
+  it("can create a question via the CreateQuestion component", () => {
     cy.intercept("POST", "/api/card").as("createCard");
 
+    cy.intercept("POST", "/api/dataset", req => {
+      // throttling to 500 Kbps (slow 3G) makes the loading indicator
+      // shows up reliably when we click on visualize.
+      req.on("response", res => {
+        res.setThrottle(500);
+      });
```
Comment (heypoom) [inline e2e/test/scenarios/embedding-sdk/create-question.cy.spec.ts:31]: I couldn't think of another way to reduce test flakiness when checking for loading indicators, let me know if you have a better idea 🙏🏻

**Review by oisincoveney — APPROVED** [review]
I think it looks good. One question - do you think we could update the type of `question` on the InteractiveQuestionContext to make it a non nullable property with these changes?

**Comment by oisincoveney** [comment]
> @oisincoveney I think the `question` field can still be `undefined` when the `questionId` points to an invalid question though, currently the `question` field comes from the load question hook.
> 
> ```
> export interface LoadQuestionHookResult {
>   question?: Question;
> ``
> ```

hm.. wouldn't that still end up at the `SdkError` state rather than the actual component?

We don't have to do anything about this in this PR but I really would like to sort that out at some point so we can have some nicer typing for our components

**Comment by heypoom** [comment]
@oisincoveney Good point. I created a task https://github.com/metabase/metabase/issues/50456 for lifting the SdkError into InteractiveQuestionProvider itself, so we can guarantee that question will never be undefined.

**Pull request merged**